test/metabase/models/data_permissions/graph_test.clj

```diff
@@ -235,3 +240,176 @@
+      {:perms/download-results :ten-thousand-rows} {:download {:schemas :limited}}
+      {:perms/download-results :no}                {}
+      {:perms/manage-table-metadata :yes}          {:data-model {:schemas :all}}
+      {:perms/manage-table-metadata :no}           {}
+      {:perms/manage-database :yes}                {:details :yes}
+      {:perms/manage-database :no}                 {}
+      ;; with schemas:
+      {:perms/data-access
+       {"PUBLIC" {1 :unrestricted
+                  2 :no-self-service}}}            {:data {:schemas {"PUBLIC" {1 :all}}}}
+      {:perms/data-access
+       {"PUBLIC" {1 :unrestricted
+                  2 :unrestricted}}}               {:data {:schemas {"PUBLIC" :all}}}
+      {:perms/data-access
+       {"PUBLIC" {1 :no-self-service
+                  2 :no-self-service}}}            {}
+      {:perms/download-results
+       {"PUBLIC" {1 :one-million-rows
+                  2 :no}}}                         {:download {:schemas {"PUBLIC" {1 :full}}}}
+      {:perms/download-results
+       {"PUBLIC" {1 :one-million-rows
+                  2 :ten-thousand-rows}}}          {:download {:schemas {"PUBLIC" {1 :full
+                                                                                   2 :limited}}}}
+      {:perms/manage-table-metadata
+       {"PUBLIC" {1 :yes}}}                        {:data-model {:schemas {"PUBLIC" :all}}}
+      {:perms/manage-table-metadata
+       {"PUBLIC" {1 :no}}}                         {}
+      {:perms/manage-table-metadata
+       {"PUBLIC" {1 :yes
+                  2 :no}}}                         {:data-model {:schemas {"PUBLIC" {1 :all}}}}
+      ;; multiple schemas
+      {:perms/data-access
+       {"PUBLIC" {1 :unrestricted}
+        "OTHER" {2 :no-self-service}
+        "SECRET" {3 :block}}}                      {:data {:schemas {"PUBLIC" :all, "SECRET" :block}}}
+      {:perms/data-access
+       {"PUBLIC" {1 :unrestricted
+                  2 :no-self-service}
+        "OTHER" {3 :no-self-service
+                 4 :no-self-service}}}             {:data {:schemas {"PUBLIC" {1 :all}}}})))
+
+(deftest rename-perm-test
+  (is (api-graph=
+       (#'data-perms.graph/rename-perm {:perms/data-access :unrestricted
+                                        :perms/native-query-editing :yes
+                                        :perms/manage-database :no
+                                        :perms/download-results :one-million-rows
+                                        :perms/manage-table-metadata :no})
+       {:data {:native :write, :schemas :all} :download {:native :full, :schemas :full}})))
+
+(defn constrain-graph
+  "Filters out all non `group-id`X`db-id` permissions"
+  [group-id db-id graph]
+  (-> graph
+      (assoc :groups {group-id (get-in graph [:groups group-id])})
+      (assoc-in [:groups group-id] {db-id (get-in graph [:groups group-id db-id])})))
+
+(defn- test-data-graph [group]
```

Comment:
The tests below this line were copied from `metabase.models.permissions-test` (and updated to work with the new code)